doc/guide/dynamics/dynamics-dysolve-propagator.rst

```diff
@@ -0,0 +1,123 @@
+
+.. math::
+
+ \displaystyle S^{(n)}(\boldsymbol{\omega}_n, \delta t) = \frac{-i^n}{2^n} \sum_{k^{(n)}}...\sum_{k^{(0)}} e^{-i\lambda_{k^{(n)}}\delta t}\int_{0}^{\delta t}e^{i(\boldsymbol{\omega}_n[n] + \lambda_{k^{(n)}} - \lambda_{k^{(n-1)}})t_n}dt_n ... \int_{0}^{t_2}e^{i(\boldsymbol{\omega}_n[1] + \lambda_{k^{(1)}} - \lambda_{k^{(0)}})t_1}dt_1
+
+.. math::
+
+ \displaystyle \times \bra{k^{(n)}}X \ket{k^{(n-1)}}...\bra{k^{(1)}}X \ket{k^{(0)}}\ket{k^{(n)}}\bra{k^{(0)}}
+
+This assumes that :math:`H_0` and :math:`X` are written in :math:`H_0`'s basis. The nested integral can be evaluated analytically in code with integration by parts. Also, in practice, the summation over :math:`n` is truncated to the first few terms and the evolution is divided into small time increments. So, with :math:`\tau` being the time ordering operator,
+
+.. math::
+ \displaystyle U(T,0) = \tau\prod_{p=0}^{P}U((p+1)\delta t, p\delta t) = \tau\prod_{p=0}^{P}\left(\sum_{n=0}^{\infty}U^{(n)}((p+1)\delta t, p\delta t)\right)
+
+.. math::
+ \displaystyle \approx  \tau\prod_{p=0}^{P}\left(\sum_{n=0}^{r}U^{(n)}((p+1)\delta t, p\delta t)\right)
+
+As we can see, computing :math:`S^{(n)}(\boldsymbol{\omega}_n, \delta t)` for a given range on :math:`n` once allows us to calculate as many propagators as we want because this quantity depends on the time increment :math:`\delta t` and not the current time in the evolution. Therefore, in the long run and with the right ressources, Dysolve can be an extremely useful and efficient method for calculating propagators of systems with the form described at the beginning of this section.
+
+Zeroth order example
+====================
+As an example, let's find what :math:`U^{(0)}(t+\delta t, t)` is. First, 
+
+.. math::
+
+ \displaystyle S^{(0)}(\boldsymbol{\omega}_0, \delta t) = \sum_{k^{(0)}}e^{-i\lambda_{k^{(0)}}\delta t} \ket{k^{(0)}}\bra{k^{(0)}} = e^{-iH_0\delta t}
+
+Finally, because :math:`\left\{\boldsymbol{\omega}_0\right\}` is empty, we get
+
+.. math::
+
+ \displaystyle U^{(0)}(t+\delta t, t) = e^{-iH_0\delta t}
+
+.. _dysolve_code_example:
+
+Code example
+============
+
+The following code shows a simple example on how to use QuTiP's implementation of Dysolve. Here, :math:`H(t) = \sigma_z + \cos(t)\sigma_x`.
+
+.. code-block:: Python
+    
+    from qutip.solver.dysolve_propagator import DysolvePropagator, dysolve_propagator
+    from qutip import sigmax, sigmaz
+
+    max_order = 5
+    omega = 1
+    t_i = 0
+    t_f = 1
+    dt = 0.1
+    H_0 = sigmaz()
+    X = sigmax()
+
+    #Initialize and call to compute the propagators for each time increment t => U(t + dt, t)
+    dysolve = DysolvePropagator(
+                max_order, H_0, X, omega
+            )
+    dysolve(t_i, t_f, dt)
+    Us = dysolve.Us
```

Comment:
dysolve is too close to our solve function and could be added as such later, so let's use another variable.
Also the call return the propagator.
```suggestion
    dy_U = DysolvePropagator(
        max_order, H_0, X, omega, options={"max_dt": dt}
    )
    Us = dy_U(t_i, t_f, dt)
```